Add context menu option to list programs in a series

mlehtima opened the pull request:
Requires https://github.com/joonne/harbour-nayttamo/pull/24

mlehtima committed:
Add context menu option to list programs in a series.

Remove some debug printing. Simplify API code.

joonne commented on qml/js/yleApi.js:
```diff
@@ -108,7 +108,7 @@ var api = (function(appId, appKey) {
                                : "",
                  episodeNumber: program.episodeNumber ? program.episodeNumber : "",
-                 seriesId: program.partOfSeason && program.partOfSeason.id,
+                 seriesId: program.partOfSeries && program.partOfSeries.id
```
so `partOfSeason` is then something that we can use later on and find programs by season?

mlehtima replied:
We already use `program.partOfSeason.seasonNumber `a few lines above, this was probably a bug.

joonne replied:
:+1:

joonne commented on qml/pages/ProgramsPage.qml:
```diff
@@ -7,62 +7,69 @@ Page {
-            })
+        if (category.id) {
+            YleApi.getProgramsByCategoryId(category.id, limit, offset)
+                .then(function(programs) {
```
extract this function since it is duplicated here and in the alternative branch?

mlehtima replied:
Good point, I'll fix that

joonne replied:
:+1:

mlehtima committed:
Fix dynamic loading in search page and improve appearance.

mlehtima committed:
Center placeholder text in main page.

mlehtima commented:
I made the search field get automatic focus so the virtual keyboard would appear automatically, wondering if we want to have that.

joonne commented:
We can try that, I kinda think that the self appearing keyboard is annoying and the placeholder text "invites" the user to do the search. No strong opinions though :)

joonne commented on qml/js/yleApi.js:
```diff
@@ -126,6 +126,14 @@ var api = (function(appId, appKey) {
      }
 
+     function getProgramsBySeriesId(seriesId, limit, offset) {
+         var url = apiUrl + programsUrl + '&series=' + seriesId + '&availability=ondemand&mediaobject=video&offset=' + offset + '&limit=' + limit;
```
do we need the contentprotection for this also: `&contentprotection=22-0,22-1&availability=ondemand` ?

mlehtima replied:
Yes, it should be there, I will move those common parameters to the `programsUrl` variable.

mlehtima committed:
Also check that publication event has available media.

mlehtima commented:
I removed the focus change because it caused some overlap with keyboard. I also added a check to make sure publication event has media available to minimize risk of non-working videos.

mlehtima commented:
I added also series listing context menu to first page of the app for current broadcasts. Would you have time to review this so we could make a new release early next week? The current version in store is some much worse.

joonne commented:
I'll review and test this during next week for sure, I have a deadline for one other hobby project for Wednesday but it should be ready today/tomorrow. Then I should have more time for näyttämö :)

joonne commented on qml/js/yleApi.js:
```diff
@@ -8,7 +8,7 @@ var api = (function(appId, appKey) {
      var credentials = "app_id=" + appId + "&app_key=" + appKey;
 
-     var programsUrl = "/programs/items.json?" + credentials;
+     var programsUrl = "/programs/items.json?" + credentials + "&contentprotection=22-0,22-1&availability=ondemand&mediaobject=video";
```
:+1:

joonne replied:
:+1:

joonne commented on qml/js/yleApi.js:
```diff
@@ -26,7 +26,7 @@ var api = (function(appId, appKey) {
          return programs.reduce(function(acc, program) {
              var foundPublicationEvents = program.publicationEvent.filter(function(event) {
-                 return event.temporalStatus === "currently" && event.type === "OnDemandPublication";
+                 return event.temporalStatus === "currently" && event.type === "OnDemandPublication" && event.media && event.media.available;
```
This was missing also? :thinking:

mlehtima replied:
I saw some videos being listed without a event.media value and some without available media.

joonne commented on qml/pages/FirstPage.qml:
```diff
@@ -62,6 +62,14 @@ Page {
                         }
                     }
+                    MenuItem {
+                        visible: modelData.seriesId ? true : false
```
useless ternary but could be replaced with `Boolean(modelData.seriesId)`?

mlehtima replied:
Changed this and another similar line to use Boolean function.

joonne reviewed (approved):
Looks good! The new functionality with showing programs in series is amazing, good job! :)

joonne committed:
Add translations.

joonne commented:
As a general note we could think of adding a spinner or some other indication when the autoloading new items, I'll add a ticket for it.

mlehtima committed:
Remove unused often failing API call.

mlehtima commented:
A spinner would probably be a good idea. I pushed the boolean changes and also removed one unused API call from ProgramOverviewPage.

joonne commented on qml/pages/ProgramOverviewPage.qml:
```diff
@@ -8,10 +8,6 @@ Page {
 
     property var program: ({})
 
-    Component.onCompleted: {
```
How was this even there anymore :sweat_smile:

joonne commented:
looks good, ready to merge :)

mlehtima commented:
I think now would be a good time to make a new release. Could you do it when you have time. Check also the issues about supported Sailfish OS versions in store and also x86 build would be nice. I just tested the app on Jolla Tablet and it worked well. If you want me to test the x86 build you make send the rpm via email.
